contrib/platform/src/com/sun/jna/platform/mac/CoreFoundationUtil.java

```diff
@@ -0,0 +1,142 @@
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.mac;
+
+import static com.sun.jna.platform.mac.CoreFoundation.kCFStringEncodingUTF8;
+
+import java.util.Collection;
+
+import com.sun.jna.Memory;
+import com.sun.jna.platform.mac.CoreFoundation.CFAllocatorRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFNumberType;
+import com.sun.jna.platform.mac.CoreFoundation.CFTypeRef;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+
+/**
+ * Provides utilities for Core Foundations
+ */
+public class CoreFoundationUtil {
+
+    private static final CoreFoundation CF = CoreFoundation.INSTANCE;
+
+    public static final CFAllocatorRef ALLOCATOR = CF.CFAllocatorGetDefault();
+
+    private CoreFoundationUtil() {
+    }
+
+    /**
+     * Convert a reference to a Core Foundations LongLong into its {@code long}
+     *
+     * @param theLong
+     *            The pointer to a 64-bit integer
+     * @return The corresponding {@code long}
+     */
+    public static long cfPointerToLong(CFTypeRef theLong) {
+        LongByReference lbr = new LongByReference();
+        CF.CFNumberGetValue(theLong, CFNumberType.kCFNumberLongLongType.ordinal(), lbr);
+        return lbr.getValue();
+    }
+
+    /**
+     * Convert a reference to a Core Foundations Int into its {@code int}
+     *
+     * @param p
+     *            The pointer to an integer
+     * @return The corresponding {@code int}
+     */
+    public static int cfPointerToInt(CFTypeRef p) {
+        IntByReference ibr = new IntByReference();
+        CF.CFNumberGetValue(p, CFNumberType.kCFNumberIntType.ordinal(), ibr);
+        return ibr.getValue();
+    }
+
+    /**
+     * Convert a reference to a Core Foundations Boolean into its {@code boolean}
+     *
+     * @param cfBoolean
+     *            The pointer to a boolean
+     * @return The corresponding {@code boolean}
+     */
+    public static boolean cfPointerToBoolean(CFTypeRef cfBoolean) {
+        return CF.CFBooleanGetValue(cfBoolean);
+    }
+
+    /**
+     * Convert a reference to a Core Foundations String into its
+     * {@link java.lang.String}
+     *
+     * @param cfTypeRef
+     *            The pointer to a CFString
+     * @return The corresponding {@link java.lang.String}
+     */
+    public static String cfPointerToString(CFTypeRef cfTypeRef) {
```

Comment:
Should the parameter be a CFStringRef?

Reply:
Another of the same type safety/casting decisions discussed for `CFNumberRef` and `CFBooleanRef`.